src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcs.stepsdesign;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.StorageException;
+import io.cdap.e2e.pages.actions.CdfGcsActions;
+import io.cdap.e2e.pages.actions.CdfStudioActions;
+import io.cdap.e2e.pages.locators.CdfGCSLocators;
+import io.cdap.e2e.pages.locators.CdfStudioLocators;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.CdfHelper;
+import io.cdap.e2e.utils.ConstantsUtil;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.e2e.utils.StorageClient;
+import io.cdap.plugin.common.stepsdesign.TestSetupHooks;
+import io.cdap.plugin.utils.E2ETestConstants;
+import io.cucumber.java.en.Then;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import stepsdesign.BeforeActions;
+
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * GCS Plugin related common step design.
+ */
+public class GCSBase implements CdfHelper {
+
+  static {
+    SeleniumHelper.getPropertiesLocators(CdfStudioLocators.class);
+  }
+
+  @Then("Enter GCS property projectId and reference name")
+  public void enterGCSPropertyProjectIdAndReferenceName() throws IOException {
+    CdfGcsActions.enterProjectId();
+    CdfGcsActions.enterReferenceName();
+  }
+
+  @Then("Enter GCS property path {string}")
+  public void enterGCSPropertyPath(String path) {
+    CdfGcsActions.getGcsBucket(PluginPropertyUtils.pluginProp(path));
+  }
+
+  @Then("Select GCS property format {string}")
+  public void selectGCSPropertyFormat(String format) throws InterruptedException {
+    CdfGcsActions.selectFormat(format);
+  }
+
+  @Then("Enter GCS property delimiter {string}")
+  public void enterGCSPropertyDelimiter(String delimiter) {
+    CdfGcsActions.enterDelimiterField(PluginPropertyUtils.pluginProp(delimiter));
+  }
+
+  @Then("Close the GCS properties")
+  public void closeTheGCSProperties() {
+    CdfGcsActions.closeButton();
+  }
+
+  @Then("Enter the GCS properties with blank property {string}")
+  public void enterTheGCSPropertiesWithBlankProperty(String property) throws InterruptedException {
+    if (!property.equalsIgnoreCase("referenceName")) {
+      CdfGcsActions.enterReferenceName();
+    }
+    if (!property.equalsIgnoreCase("path")) {
+      CdfGcsActions.getGcsBucket("gs://cdf-athena-test/dummy");
```

Comment:
bucket path should be a parameter, don't hardcode it

Reply:
This dummy path is hardcoded here just to validate mandatory property error. Otherwise we are creating buckets at runtime before test scenario and passing that path to the tests.